### DISCUSSION TITLE
Custom CSS Light/Dark Support

### DISCUSSION THREAD

**Discussion by kevinzakka** [pradyunsg/furo#485] [Questions] [answered]
I have a custom CSS override file that is correctly displayed by `furo` (which btw is a beautiful theme, thanks!) in light mode. I was wondering if there is an easy way for me to specify for each component in this CSS file, a dark theme variant? For example:

```css
.at {
  color: darkred;
}

.at[data-theme="dark"] {
  color: red;
}
```

I only got the above to work for `body`  by checking inspect but it doesn't work for other vars. Would appreciate some help / clarification!

**Comment by dgw** [comment]
Do you mean `body[data-theme="dark"]` works as you want? If so, just use that as a parent selector: `body[data-theme="dark"] div.somethingelse` for example.

**Comment by pradyunsg** [comment, marked as answer]
The only supported way of doing this is via CSS variables.

If you have custom CSS, I'd still recommend having CSS variables for all the things you wanna change (between dark vs light) and using https://pradyunsg.me/furo/customisation/#light-css-variables-dark-css-variables to change them.

So... for your case, I'd do:

```css
.at {
  color: var(--my-color);
}
```

```py
html_theme_options = {
    "light_css_variables": {
        "my-color": "darkred",
    },
    "dark_css_variables": {
        "my-color": "red",
    },
}
```

For reasons related to supporting the dark theme on no-js browsers, Furo has a slightly quirky approach for managing the dark mode, which requires [duplicating those declarations](https://github.com/pradyunsg/furo/blob/8eba6499b812d7aeab2a99fcdf33e8bcb07b05fc/src/furo/theme/furo/partials/_head_css_variables.html#L14).

**Reply by kevinzakka** [reply]
Apologies for the delay, this worked! Thanks for being so responsive :)
